kong/global.lua

```diff
@@ -201,6 +201,33 @@ function _GLOBAL.init_worker_events()
     return nil, err
   end
 
+  local PAYLOAD_MAX_LEN, PAYLOAD_TOO_BIG_ERR = 65535, "failed to publish event: payload too big"
+  local TRUNCATED_PREFIX = ", truncated payload: "
+  local DEFAULT_TRUNCATED_PAYLOAD = TRUNCATED_PREFIX .. "not a serialized object"
+  local LEN = (#PAYLOAD_TOO_BIG_ERR + #DEFAULT_TRUNCATED_PAYLOAD) * 2
+
+  -- There is a limit for the payload size that events lib allows to send,
+  -- we overwrite `post` method to truncate the payload and send it again
+  -- when we get error message: "payload too big"
+  local native_post = worker_events.post
+  worker_events.post = function (source, event, data, unique)
+    local ok, err = native_post(source, event, data, unique)
+    -- exceeds the upper limit for the size of the payload
+    if err == PAYLOAD_TOO_BIG_ERR then
+      if type(data) == "string" then
+        -- truncate the payload and send it again  
+        data = PAYLOAD_TOO_BIG_ERR .. ", truncated payload: " .. string.sub(data, 1, PAYLOAD_MAX_LEN - LEN)
```

Comment:
Should we send this too-big truncated payload? 
Since it is truncated, could it be `string.sub(data, 1, 1024)` or something else?

Reply:
It's a valuable thinking for the length. In my feel, any length to be truncated is not convincing enough except that we try our best and  keep the content as enough as we can or we can just tell the fact that the payload is truncated without any information about the original payload. And I think `try our best` would be better.
So that's my opinion, WDYT?

Reply:
Could the event receiver do something with this truncated data? if not, I think it may not make sense.